ewatercycle/models/lisflood.py

```diff
@@ -241,6 +241,53 @@ def get_value_as_xarray(self, name: str) -> xr.DataArray:
 
         return da
 
+    def _coords_to_indices(self, name: str, lat: Iterable[float], lon: Iterable[float]) -> Iterable[int]:
+        """Convert lat, lon coordinates into model indices."""
+        grid_id = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid_id) # shape returns (len(y), len(x))
+        x_model = self.bmi.get_grid_x(grid_id)
+        y_model = self.bmi.get_grid_y(grid_id)
+        spacing_model = self.bmi.get_grid_spacing(grid_id)
+
+        # Create a grid from coordinates
+        x_vectors, y_vectors = np.meshgrid(x_model, y_model)
+
+        indices = []
+        # in lisflood, x corresponds to lon, and y to lat.
+        # this might not be the case for other models!
+        for x, y in zip(lon, lat):
+            # here we use Euclidean distance, but it is not accurate as the coordinates are in lon/lat and degrees.
```

Comment:
Probably fine like this, but you could consider making it more accurate with this simple conversion: 

lon [km] = lon [deg] * 111 [km / deg]
lat [km] = lat [deg] * 111 [km / deg] * cos( lon [deg] )

https://stackoverflow.com/a/39540339

Reply:
oh, thanks for the links. I used a similar equation mentioned here https://en.wikipedia.org/wiki/Geographical_distance. 
Meantime, I don't know why you reversed the `lat` and `lon` in your sample codes. I assume it was by mistake.